### PR TITLE
Simple type

### PR DESCRIPTION
**Pull request by RobertGemmaJr**
- Add W3D base class 
- Add `Color` and `Vector3` unity class types to the XMl class structure with custom getters/setters
- Adds `[Serializable` tag for all classes
- Adds `vs-classes.cs` for reference

`vs-classes.cs` is a autogenerated class file like `Classes.cs` but done directly in Visual Studio. It was done with `caveschema.xsd` (instead of `everything.xml`. Neither are perfect and I continue my work in `Classes.cs` but I reference both often

### CODE REVIEW

File: unity/CAVE/Assets/W3D_Translator/Classes.cs [hunk 1]
```diff
@@ -8,11 +11,13 @@
...
 namespace W3D
 {
+
     /********** STORY            ***********/
 
 
+    [Serializable]
     [XmlRoot(ElementName="Story")]
-    public class Story
+    public class Story : W3D
```
Comment (broarr) [inline unity/CAVE/Assets/W3D_Translator/Classes.cs:20]: What does this `: W3D` do?

Comment (broarr) [reply unity/CAVE/Assets/W3D_Translator/Classes.cs:20]: Ahh `W3D` is a base class. I'd consider moving the base class to the top of this file before the rest of the classes that inherit from this. I don't know C# or C++ style guides, but is it common to name a class the same name as the namespace? That caused me a little confusion

File: unity/CAVE/Assets/W3D_Translator/Classes.cs [hunk 2]
```diff
@@ -1516,4 +1805,37 @@ public class Disc
...
+        static public Vector3 ConvertVector3(string vectorString)
+        {
+            string[] strings=vectorString.Trim(new Char[] { ' ', '(', ')' }).Split(",");
+            return new Vector3(
+                float.Parse(strings[0]),
+                float.Parse(strings[1]),
+                float.Parse(strings[2])
+            );
+        }
+    }
```
Comment (broarr) [inline unity/CAVE/Assets/W3D_Translator/Classes.cs:1840]: I get the pretty print, that's a nice thing to have on all child classes. These static methods seem unrelated to their underlying data. Does a camera really need `ConvertColor`? Does a color really need `ConvertVector3`? Can you make these util functions within the same namespace or something? Or do they need to be wrapped in a class structure?

File: vs-classes.cs [hunk 3]
```diff
@@ -0,0 +1,4369 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by a tool.
+//     Runtime Version:4.0.30319.42000
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
```
Comment (broarr) [inline vs-classes.cs:9]: As a rule I don't commit auto-generated files to a repository, but I will commit a makefile or build script that generates those files. Is this vs-classes file something that can be easily generated at build time? Is there a good reason to include this autogenerated file in the repo?

Comment (broarr) [reply vs-classes.cs:9]: The reason I don't commit autogenerated code is to discourage people from editing files. If the file is version controlled people are more likely to edit it and break it

Comment (broarr) [reply vs-classes.cs:9]: How does this code relate to `Classes.cs`? This is the xml parser right?

Comment (broarr) [reply vs-classes.cs:9]: Also, where is this file used?

**Review by broarr — CHANGES_REQUESTED** [review]
If possible, I'd remove the `vs-classes.cs` file from the repo, and provide a script to generate the `vs-classes.cs` file. I'm kinda confused as to what `vs-classes.cs` does and how it's called. I didn't see it imported in your `W3D_Translator` project

File: unity/CAVE/Assets/W3D_Translator/Classes.cs (hunk 1 above)
Comment (RobertGemmaJr) [reply unity/CAVE/Assets/W3D_Translator/Classes.cs:20]: Okay! I moved it and renamed it as `Xml`

File: unity/CAVE/Assets/W3D_Translator/Classes.cs (hunk 2 above)
Comment (RobertGemmaJr) [reply unity/CAVE/Assets/W3D_Translator/Classes.cs:1840]: Everything needs to be inside of a class in c#... I could add a methods class? Or just add them inside the `CLI` class in `CLI.cs`? That file is going to get pretty messy though so I'm not sure what's best

File: vs-classes.cs (hunk 3 above)
Comment (RobertGemmaJr) [reply vs-classes.cs:9]: I'm just using it for reference file and editing `Classes.cs` so, really, people don't need a script to re-generate the file either. I added it to the gitignore, that seems reaosnable?

File: unity/CAVE/Assets/W3D_Translator/Classes.cs (hunk 2 above)
Comment (broarr) [reply unity/CAVE/Assets/W3D_Translator/Classes.cs]: Eh, I'd just keep them in here. I was mostly curious

File: vs-classes.cs (hunk 3 above)
Comment (broarr) [reply vs-classes.cs:9]: Sounds good to me!

**Review by broarr — APPROVED** [review]

**Comment by broarr** [comment]
Looking good!

File: vs-classes.cs (hunk 3 above)
Comment (RobertGemmaJr) [reply vs-classes.cs:9]: Yeah I mean they do pop up in intellisense on all of the classes but it's not much of a bother
